### PR TITLE
Feature/windows clang

### PR DESCRIPTION
**Pull request by Janrupf**
This makes the windows CI build use clang cl. Currently it doesn't build because of some compilation error, but it now uses clang-cl and Ninja on windows.

### CODE REVIEW

**Review by DethRaid — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -90,35 +90,80 @@ jobs:
...
           Start-Process .\VulkanSDK-Installer.exe /S -Wait
+      # TODO: Try cache that, but where the fuck does it get installed?
       # Download the Vulkan Runtime
       - name: Download Vulkan Runtime
         shell: bash
         run: |
           curl -L \
            'https://sdk.lunarg.com/sdk/download/${{ matrix.vulkan }}/windows/vulkan-${{ matrix.vulkan }}-runtime.exe' -o VulkanRuntime-Installer.exe
-          ls 
+          ls
```
Comment (cwfitzgerald) [inline .github/workflows/ci.yml:129]: Left in a `ls`

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]

**Comment by DethRaid** [comment]
Also really need to add `-Wno-documentation` to the build args

**Comment by DethRaid** [comment]
nah

**Pull request closed without merging**